src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs

```diff
@@ -0,0 +1,402 @@
+            }
+        }
+
+        IgnoreAppResourceDictionaryChanges = false;
+
+        // _currentUseLightMode = useLightColors;
+        // _currentFluentThemeResourceUri = fluentThemeResourceUri;
+        // _currentAccentColor = AccentColorHelper.GetAccentColor();
+    }
+
+    internal static void OnAppResourcesChanged()
+    {
+        if(IgnoreAppResourceDictionaryChanges) return;
+
+        if(AppResourceContainsFluentDictionary(out string theme))
+        {
+            Application.Current.Theme = theme;
+        }
+        else
+        {
+            Application.Current.Theme = "None";
+        }
+
+    }
+
+    internal static void OnWindowThemeChanged(Window window, string oldTheme, string newTheme)
+    {
+        // Redundant
+        if(window == null) return;
+
+        if(newTheme == "None")
+        {
+            RemoveFluentFromWindow(window);
+            return;
+        }
+
+        ApplyFluentOnWindow(window);
+    }
+
+    internal static void LoadDeferredApplicationTheme()
+    {
+        if(_deferThemeLoading)
+        {
+            _deferThemeLoading = false;
+            OnApplicationThemeChanged("None", Application.Current.Theme);
+        }
+    }
+
+    internal static bool VerifyApplicationTheme(string theme)
+    {
+        return theme == "System" || theme == "Light" || theme == "Dark" || theme == "None";
+    }
+
+    internal static bool ApplyStyleOnWindow(Window window)
+    {
+        if(!IsFluentThemeEnabled && window.Theme == "None") return false;
+
+        bool useLightColors = GetUseLightColors(Application.Current.Theme);
+        if(window.Theme != "None")
+        {
+            useLightColors = GetUseLightColors(window.Theme);
+        }
+        ApplyStyleOnWindow(window, useLightColors);
+        return true;
+    }
+
+    #region Internal Properties
+    internal static bool IsFluentThemeEnabled
+    {
+        get
+        {
+            if(Application.Current == null) return false;
+            return Application.Current.Theme != "None";
+        }
+    }
+    internal static bool IgnoreAppResourceDictionaryChanges
+    {
+        get
+        {
+            return _ignoreAppResourceDictionaryChanges;
+        }
+        set
+        {
+            _ignoreAppResourceDictionaryChanges = value;
+        }
+    }
+
+    internal static double DefaultFluentThemeFontSize => 14;
+    
+    #endregion
+
+    #region Private Methods
+
+    private static void RemoveFluentFromApplication()
+    {
+        if(Application.Current == null) return;
+
+        int index = FindLastFluentThemeResourceDictionaryIndex(Application.Current.Resources);
+        
+        if(index != -1)
+        {
+            Application.Current.Resources.MergedDictionaries.RemoveAt(index);
+        }
+
+        foreach(Window window in Application.Current.Windows)
+        {
+            if(!_fluentEnabledWindows.HasItem(window))
+            {
+                RemoveStyleOnWindow(window);
```

Comment:
This is `RemoveFluentFromApplication` but you call `RemoveStyleOnWindow` rather than `RemoveFluentFromWindow`. If that is desired, it would be helpful to justify it in a comment.